src/algoliasearch.helper.js

```diff
@@ -306,10 +305,15 @@ AlgoliaSearchHelper.prototype.searchForFacetValues = function(facet, query, maxF
 
     content = Array.isArray(content) ? content[0] : content;
 
-    content.facetHits = forEach(content.facetHits, function(f) {
-      f.isRefined = isDisjunctive ?
-        state.isDisjunctiveFacetRefined(facet, f.value) :
-        state.isFacetRefined(facet, f.value);
+    content.facetHits = Array.isArray(content.facetHits)
+      ? content.facetHits
+      : [];
+
+    content.facetHits = content.facetHits.map(function(f) {
```

Comment:
We should have stick with `forEach` since we perform a side effect inside the loop. The return value is useless in that case. We mutate the original ref of the value.

Reply:
You're right, this should be forEach

Reply:
https://github.com/algolia/algoliasearch-helper-js/pull/686